swarm.go

```diff
@@ -22,9 +22,16 @@ type SwarmConfig struct {
 	// EnableAutoRelay enables the "auto relay user" feature.
 	// Node will find and use advertised public relays when it determines that
 	// it's not reachable from the public internet.
+	//
+	// Deprecated: This flag is deprecated and is overriden by
+	// `Swarm.AutoRelay.Enabled` if specified.
 	EnableAutoRelay bool
 
-	// RelayService.* controls the "auto relay service" feature.
+	// AutoRelay controls the "auto relay service" feature.
+	// When enabled, the node will use relays if it is not publicly reachable.
+	AutoRelay AutoRelay
```

Comment:
I am worried about ambiguity, better to be very clear around relay "service" and relay "client". 
 Would it be ok if we rename `AutoRelay` to more explicit `RelayClient`? 
 
```suggestion
	RelayClient RelayClient
```

This way, we will have `RelaySevice` and `RelayClient` sections that are easier to wrap the head around without reading docs.

Reply:
I'm ok with this. @vyzo, wdyt?

Reply:
yeah, fine by me.

Reply:
Applied  change in https://github.com/ipfs/go-ipfs-config/pull/154/commits/11644afd3c4e800716e8260357d5caa48fdfe90f